### ISSUE TITLE
Serializer changing underscores to hyphens

### ISSUE THREAD

**Issue by sporto** [AgilionApps/ja_serializer#4]
My serializers looks like

```
serialize "users_teams" do
	location "/users_teams/:id"

	attributes [:cid, :user_id, :team_id]
end
```

But JSON data comes out as:

```
...
"attributes": {
	"user-id": 1,
	"team-id": 1,
	"cid": null
}
...
```

I would expect the serializer to output underscores i.e. `user_id`

**Comment by alanpeabody** [comment]
@sporto I have initially gone with underscores because that is the recommendation http://jsonapi.org/recommendations/#naming.

It is however only a recommendation and not a part of the format, so I open to making this a config option to support snake_case and camelCase if there is need.

I will look at this for an upcoming 0.2.0 release.

**Comment by beerlington** [comment]
FWIW jsonapi-resources [defaults to dashes](https://github.com/cerebris/jsonapi-resources#configuration) but lets you override with underscores or camel case.

```ruby
JSONAPI.configure do |config|
  #:underscored_key, :camelized_key, :dasherized_key, or custom
  config.json_key_format = :dasherized_key
```

**Comment by alanpeabody** [comment]
Yup, that is similar to what I am thinking as well.

**Comment by alanpeabody** [comment]
@sporto I have just released version 0.1.0 to hex that adds underscore support via config. Simply update and add this to your config.exs:

```elixir
config :ja_serializer,
  key_format: :underscored
```

**Comment by sporto** [comment]
Excellent, thanks!
